tests/integration/security/trust_domain_alias/trust_domain_alias_client_test.go

```diff
@@ -0,0 +1,289 @@
+			//if err := ik.WaitForConfigs(testNS.Name(), policy); err != nil {
+			//	// Continue anyways, so we can assess the effectiveness of using `istioctl wait`
+			//	ctx.Logf("warning: failed to wait for config: %v", err)
+			//	// Get proxy status for additional debugging
+			//	s, _, _ := ik.Invoke([]string{"ps"})
+			//	ctx.Logf("proxy status: %v", s)
+			//}
+			//ctx.Logf("Sleep for 90 secs....")
+			//time.Sleep(time.Second * 90)
+
+
+			// making subtests but getting the same results
+			//t.Run("trust-domain-alias-client-validation", func(t *testing.T) {
+			//	for _, dest := range []echo.Instance{serverNakedFoo, serverNakedBar} {
+			//		src := client
+			//		dest := dest
+			//		opts := echo.CallOptions{
+			//			Target:   dest,
+			//			PortName: httpMTLS,
+			//			Host:     dest.Config().Service,
+			//			Scheme:   scheme.HTTPS,
+			//		}
+			//		ctx.NewSubTest("server:" + dest.Config().Service).
+			//			Run(func(ctx framework.TestContext) {
+			//				checker := connection.Checker{
+			//					From:          src,
+			//					Options:       opts,
+			//					ExpectSuccess: false,
+			//				}
+			//				checker.CheckOrFail(ctx)
+			//			})
+			//	}
+			//})
+
+			verify := func(t *testing.T, src echo.Instance, dest echo.Instance, s scheme.Instance, allow bool) {
+				t.Helper()
+				want := "allow"
+				if !allow {
+					want = "deny"
+				}
+				name := fmt.Sprintf("server:%s[%s]", dest.Config().Service, want)
+				t.Run(name, func(t *testing.T) {
+					t.Helper()
+					opt := echo.CallOptions{
+						Target:   dest,
+						PortName: httpMTLS,
+						Host:     dest.Config().Service,
+						Scheme:   s,
+					}
+
+					// Use checker to test
+					//src := client
+					checker := connection.Checker{
+						From:          src,
+						Options:       opt,
+						ExpectSuccess: allow,
+					}
+					checker.CheckOrFail(ctx)
+
+					// directly use echo.Call
+					//retry.UntilSuccessOrFail(t, func() error {
+					//	var resp client2.ParsedResponses
+					//	var err error
+					//	resp, err = src.Call(opt)
+					//	//resp, err = src.Call(opt)
+					//	if allow {
+					//		if err != nil {
+					//			return fmt.Errorf("want allow but got error: %v", err)
+					//		} else if err := resp.CheckOK(); err != nil {
+					//			return fmt.Errorf("want allow but got %v: %v", resp, err)
+					//		}
+					//	} else {
+					//		if err == nil {
+					//			return fmt.Errorf("want deny but got allow: %v", resp)
+					//			//return nil
+					//		}
+					//		// Look up for the specific "tls: unknown certificate" error when trust domain validation failed.
+					//		if tlsErr := "tls: unknown certificate"; !strings.Contains(err.Error(), tlsErr) {
+					//			return fmt.Errorf("want error %q but got %v", tlsErr, err)
+					//		}
+					//	}
+					//	return nil
+					//}, retry.Delay(250*time.Millisecond), retry.Timeout(30*time.Second), retry.Converge(5))
+				})
+			}
+
+			verify(t, client, serverNakedBar, scheme.HTTPS, false)
```

Comment:
Change this line's scheme to `scheme.HTTP` seems would fix the issue on my end.
You can then enable the verify below. And you will consistently get passing for both foo and bar.